src/Data/Enum/Gen.purs

```diff
@@ -0,0 +1,16 @@
+module Data.Enum.Gen where
+
+import Prelude
+
+import Control.Monad.Gen (class MonadGen, chooseInt)
+import Data.Enum (class BoundedEnum, fromEnum, toEnum)
+import Partial.Unsafe (unsafePartial)
+import Data.Maybe (fromJust)
+
+-- | Create a random generator for a finite enumeration. `toEnum i` must be
+-- | well-behaved: it must return a `Just` value for all `Int`s between
+-- | `fromEnum bottom` and `fromEnum top`.
+genBoundedEnum :: forall m a. MonadGen m => BoundedEnum a => m a
+genBoundedEnum =
+  unsafePartial fromJust <<< toEnum
```

Comment:
This generator I lifted from QuickCheck actually, it was recently added there, so it suffers from any of the same problems that that one has.

Can you explain what you're thinking with `oneOf (bottom .. top)`, I'm not sure I follow?